client/src/components/AddTodo.jsx

```diff
@@ -0,0 +1,78 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import axios from "axios";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const AddTodo = () => {
+  const BASE_URL = "http://localhost:8000/api/todo";
+
+  const navigate = useNavigate();
+
+  const [todo, setTodo] = useState("");
+  const [priority, setPriority] = useState("Low");
+
+  const queryClient = useQueryClient();
+
+  const addTodo = async (newTodo) => {
```

Comment:
too specific component , its more like a feature snippet , component needs to be generic so that it can be used later 

component should only hv details related to what to be displayed coming from props nad if any actions are associated to it they should also be in props